hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterFileSystem.java

```diff
@@ -319,19 +320,22 @@ void checkTempDir(final Path tmpdir, final Configuration c, final FileSystem fs)
         HFileArchiver.archiveRegions(c, fs, this.rootdir, tableDir,
```

Comment:
The archiveRegions  will archive all the hfiles and if all successfull... Then it will remains some dirs such as  <rootdir>/.tmp/data/ns/table ?   If we've deleted a namespace or table , the ns or table dir under .tmp will be also removed   (If no, the we will remain many useless dir, that's not good) ? 
Another concern is :  if hfiles archiving failure happen and enable the snapshot acl sync,   will we remain much hfiles under the .tmp dir ?  will be storage consuming if so... maybe we need a UT ?

Reply:
The tmp namespace or table dirs are deleted if namespace or table  are dropped. There are UTs in TestSnapshotScannerHDFSAclController to check it.
Add a warn log to record unarchived regions, if this problem happens in practice, we can fix this? And add UT in TestSnapshotScannerHDFSAclController.